fix(skeleton-text): animation no longer jumps on large skeleton text elements

danielehrhardt opened the pull request:
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ X] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ X] Build (`npm run build`) was run locally and any changes were pushed
- [X ] Lint (`npm run lint`) has passed locally and any fixes were made for failures


## Pull request type

<!-- Please do not submit updates to dependencies unless it fixes an issue. --> 

<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. --> 

Please check the type of change your PR introduces:
- [X ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 


## What is the current behavior?
https://github.com/ionic-team/ionic-framework/issues/22694

Issue Number: #22694


## What is the new behavior?
- Smooth Animation even if the width is more then 800px

## Does this introduce a breaking change?

- [ ] Yes
- [X ] No

<!-- If this introduces a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

<!-- Any other information that is important to this PR such as screenshots of how the component looks before and after the change. -->

liamdebeasi commented on core/src/components/skeleton-text/skeleton-text.scss:
```diff
@@ -46,24 +46,31 @@ span {
+      $skeleton-text-background-animated 60%,
+      transparent
+    );
+    animation: shimmer 2s infinite;
```
```suggestion
    animation: shimmer 1s infinite;
```

The old animation was 1s long

liamdebeasi reviewed (changes requested):
Thanks for the PR! Two things I noticed:

1. The animation should be 1s, not 2s.
2. The "shimmer" seems to be larger than it was before.

I attached a comparison below (you may need to open the GIFs in a new tab to see the full size). The left GIF has your changes, and the right GIF is what is currently in `master`.

| branch | `master` |
|--------|----------|
| ![NEW](https://user-images.githubusercontent.com/2721089/102659572-9a2c2580-4147-11eb-8309-7787d436939e.gif) | ![OLD](https://user-images.githubusercontent.com/2721089/102659583-a0ba9d00-4147-11eb-8857-2a73852d7400.gif) |

liamdebeasi commented on core/src/components/skeleton-text/skeleton-text.scss:
```diff
@@ -46,24 +46,31 @@ span {
   position: relative;
-
-  background: linear-gradient(to right, $skeleton-text-background 8%, $skeleton-text-background-animated 18%, $skeleton-text-background 33%);
-  background-size: 800px 104px;
```
Are we able to add this back? The shimmer seems to be wider as a result of this removal.

danielehrhardt commented:
I changed the animation speed. 
I also changed the background gradient. But the old one somehow look wired. 
I attached two GIFs. 
Maybe you could ask your Team, what you prefer, send me a quick comment which one you want, so i can update the PR. 
1.
![captured](https://user-images.githubusercontent.com/19570043/102719552-0c8b3a00-42ef-11eb-9ee1-1de26b5044cf.gif)
2.
![captured (1)](https://user-images.githubusercontent.com/19570043/102719558-14e37500-42ef-11eb-9eb1-ffbde7c7ab14.gif)

danielehrhardt commented:
@liamdebeasi ??

danielehrhardt commented:
@liamdebeasi Could you please check my last Question?

liamdebeasi commented:
> @liamdebeasi Could you please check my last Question?

Sorry, most of the team was out the past few weeks for winter holidays, so I didn't see your question.

Ideally this PR would keep the visual effect of the gradient exactly the same as shown in the `master` column in https://github.com/ionic-team/ionic-framework/pull/22697#pullrequestreview-555800059. Is there any reason why we can't keep the old linear-gradient we had set?

liamdebeasi commented:
Thanks, I took a look and some of the issues we discussed previously still persist (mainly the GIFs in https://github.com/ionic-team/ionic-framework/pull/22697#pullrequestreview-555800059).

I think the solution here is simpler than originally expected. The background-size of the skeleton text is set to `800px`, but the shimmer animation moves the background a total of `936px` (468 * 2). Changing the shimmer animation to go from `-400px` to `400px` seems to fix the animation on my end.

Can you give that a shot and let me know if it looks good to you? So the animation should look something like this:

```css
@keyframes shimmer {
  0% {
    background-position: -400px 0
  }

  100% {
    background-position: 400px 0
  }
}
```

If it looks good then I think we can revert the changes made here and just include the CSS Animation change.

danielehrhardt commented:
> Thanks, I took a look and some of the issues we discussed previously still persist (mainly the GIFs in [#22697 (review)](https://github.com/ionic-team/ionic-framework/pull/22697#pullrequestreview-555800059)).
> 
> I think the solution here is simpler than originally expected. The background-size of the skeleton text is set to `800px`, but the shimmer animation moves the background a total of `936px` (468 * 2). Changing the shimmer animation to go from `-400px` to `400px` seems to fix the animation on my end.
> 
> Can you give that a shot and let me know if it looks good to you? So the animation should look something like this:
> 
> ```css
> @keyframes shimmer {
>   0% {
>     background-position: -400px 0
>   }
> 
>   100% {
>     background-position: 400px 0
>   }
> }
> ```
> 
> If it looks good then I think we can revert the changes made here and just include the CSS Animation change.

In general on lager Screens the Shimmer appears two times because of the fixed Size of 800px. But it looks good to me. 
Should i change the PR or do you edit it directly?

liamdebeasi commented:
> In general on lager Screens the Shimmer appears two times because of the fixed Size of 800px.

I think this is intentional, but it might be worth revisiting the designs for this in the future.

> But it looks good to me. Should i change the PR or do you edit it directly?

If you could make the change that would be great. I do not have write access to this PR.

danielehrhardt commented:
> > In general on lager Screens the Shimmer appears two times because of the fixed Size of 800px.
> 
> I think this is intentional, but it might be worth revisiting the designs for this in the future.
> 
> > But it looks good to me. Should i change the PR or do you edit it directly?
> 
> If you could make the change that would be great. I do not have write access to this PR.

Changed the Code Back to Original with the 400px change.

danielehrhardt commented:
@liamdebeasi Could you check and merge?

liamdebeasi commented:
Merged. Thank you!
